manifests/system.pp

```diff
@@ -32,27 +32,34 @@
     }
   }
 
-  file { "${scripts_dir}/system_metrics":
-    ensure => file,
-    mode   => '0755',
-    source => 'puppet:///modules/puppet_metrics_collector/system_metrics'
+  exec { 'puppet_metrics_collector_system_daemon_reload':
+    command     => 'systemctl daemon-reload',
+    path        => ['/bin', '/usr/bin'],
+    refreshonly => true,
   }
 
   if $manage_sysstat {
     package { 'sysstat':
-      ensure => installed,
+      ensure => $system_metrics_ensure,
     }
   }
 
-  exec { 'puppet_metrics_collector_system_daemon_reload':
-    command     => 'systemctl daemon-reload',
-    path        => ['/bin', '/usr/bin'],
-    refreshonly => true,
-  }
+  if $manage_sysstat or $facts.dig('puppet_metrics_collector', 'have_sysstat') {
+    file { "${scripts_dir}/system_metrics":
```

Comment:
Nitpick, but do we need to move this `file` resource here?  IMO the file being `present` regardless of whether the `system` classes are included or not seems fine.

Reply:
There is no real reason we can't have it managed outside the `if`. The reason I put it in here is to comply with the same pattern we have for postgres and vmware collection

https://github.com/puppetlabs/puppetlabs-puppet_metrics_collector/blob/main/manifests/system.pp#L64-L68
https://github.com/puppetlabs/puppetlabs-puppet_metrics_collector/blob/main/manifests/system.pp#L74-L78

Other than that, it has a trivial benefit of removing a `puppet://` file resource that is outside of file sync, so it's metadata cannot be inlined.

Reply:
:brent_rambo: